lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -4358,10 +4358,12 @@ ConstraintResult GenericSignatureBuilder::addTypeRequirement(
                         ->getDependentType(getGenericParams());
 
       Impl->HadAnyError = true;
-      
+
       if (subjectType->is<DependentMemberType>()) {
         subjectType = resolveDependentMemberTypes(*this, subjectType);
-      } else {
+      }
+
+      if (!subjectType->is<DependentMemberType>()) {
```

Comment:
Do you think it is worth a comment here? because a `!subjectType->is<DependentMemberType>()` following an exact inverse check instead of an else may not be obvious for someone reading that later :)